In [ ]:
%pip install -q social_agents/

In [ ]:
%pip show social_agents

In [ ]:
%pip install -U pydantic
%pip install -U "langchain[together]"
%pip install -U "langchain"
%pip install -U "langgraph"

In [ ]:
import tqdm
# Reading the data
import pandas as pd

from social_agents.utils import get_st_data


for key, line in tqdm.tqdm(get_st_data("sample").items()):
    print(key)

    print(line['intervention'])
    input_text = line['intervention']
    break


In [ ]:
%%capture --no-stderr
%pip install --quiet -U langchain_openai langchain_core langchain_community tavily-python

## Prerequisites
env, LLM, ...

# Shared Task: Critical thinking generation
## Loadind Data


In [ ]:
import os, getpass
from langchain.chat_models import init_chat_model
from IPython.display import Image, display

def _set_env(var: str):
    if not os.environ.get(var):
        os.environ[var] = getpass.getpass(f"{var}: ")

_set_env("OPENAI_API_KEY")
_set_env("LANGCHAIN_API_KEY")
_set_env("MISTRAL_API_KEY")
_set_env("TOGETHER_API_KEY")

os.environ["LANGCHAIN_TRACING_V2"] = "false"
os.environ["LANGCHAIN_PROJECT"] = "shared_task_critical_questions"

#init_chat_model("meta-llama/Llama-3.2-3B-Instruct-Turbo", model_provider="together", temperature= 0.7)


In [ ]:

#llm_name = "gpt-4o-mini-2024-07-18"# "o3-mini-2025-01-31"
#llm_name_short = "gpt-4o-mini_"
llm_dic = {
    #"openai_o3mini": "o3-mini-2025-01-31",
    #"gpt-4o-mini_": "gpt-4o-mini-2024-07-18",
    "llama8b_": "meta-llama/Meta-Llama-3.1-8B-Instruct-Turbo",
    "mistral24b_": "mistral-small-2503",#"mistralai/Mixtral-8x7B-Instruct-v0.1",
    "llama70b_": "meta-llama/Llama-3.3-70B-Instruct-Turbo"

}

##  1. Zero Shot LLM to start with o3-mini

In [ ]:
llm_dic

In [ ]:

from social_agents.agent_builder import BasicCQModel


for llm_key, llm_full in llm_dic.items():
    print(f"Running {llm_key}")
    basic_agent = BasicCQModel(
        model_thread_id=f"{llm_key}_zero-shot",
        llm_name = llm_full,
        llm_num = 1,
        experiment_name= f"{llm_key}_zero-shot",
        temperature=0.7)
        
        
    display(Image(basic_agent.graph.get_graph(xray=1).draw_mermaid_png()))

# RUN
    basic_agent.run_experiment()



 12%|█▏        | 23/186 [00:41<18:22,  6.76s/it]

Iteration Elmattador__92: 20.2757 seconds


 13%|█▎        | 24/186 [00:42<13:31,  5.01s/it]

Iteration FoodAllergyMom_199_1: 0.9198 seconds
Exception repeat 1
Exception repeat 2
Exception repeat 3
Exception repeat 4


 13%|█▎        | 25/186 [00:46<12:35,  4.69s/it]

Iteration Frequent-Flyer_157: 3.9582 seconds


 14%|█▍        | 26/186 [00:46<09:22,  3.52s/it]

Iteration Glblwrmingisfak__552: 0.7742 seconds


 15%|█▍        | 27/186 [00:47<07:11,  2.71s/it]

Iteration HOLT_126: 0.8382 seconds


 15%|█▌        | 28/186 [00:48<05:41,  2.16s/it]

Iteration HOLT_239: 0.8779 seconds


 16%|█▌        | 29/186 [00:49<04:31,  1.73s/it]

Iteration HOLT_94: 0.7160 seconds


 16%|█▌        | 30/186 [00:50<03:47,  1.46s/it]

Iteration Helen__66: 0.8190 seconds
Exception repeat 1


 17%|█▋        | 31/186 [00:52<04:32,  1.76s/it]

Iteration 17th_knight__247: 2.4579 seconds


 17%|█▋        | 32/186 [00:53<04:01,  1.57s/it]

Iteration AFCHF_154: 1.1266 seconds
Exception repeat 1
Exception repeat 2


 18%|█▊        | 33/186 [00:56<04:40,  1.83s/it]

Iteration AK-traveler_186: 2.4564 seconds


 18%|█▊        | 34/186 [00:57<03:57,  1.56s/it]

Iteration AllergyDad_200: 0.9247 seconds


 19%|█▉        | 35/186 [00:57<03:26,  1.37s/it]

Iteration AngelComa__638: 0.9169 seconds


 19%|█▉        | 36/186 [00:59<03:09,  1.27s/it]

Iteration Antanagoge_104: 1.0235 seconds


 20%|█▉        | 37/186 [01:00<03:03,  1.23s/it]

Iteration Antanagoge_111: 1.1418 seconds
Exception repeat 1
Exception repeat 2
Exception repeat 3


 20%|██        | 38/186 [01:23<19:23,  7.86s/it]

Iteration Antanagoge_227: 23.3346 seconds


 21%|██        | 39/186 [01:43<28:24, 11.60s/it]

Iteration Antanagoge_239: 20.3133 seconds


 22%|██▏       | 40/186 [01:44<20:23,  8.38s/it]

Iteration Bill_106: 0.8777 seconds
Exception repeat 1


 22%|██▏       | 41/186 [01:46<15:35,  6.45s/it]

Iteration CF_29: 1.9488 seconds


 23%|██▎       | 42/186 [01:47<11:30,  4.79s/it]

Iteration CF_46: 0.9231 seconds


 23%|██▎       | 43/186 [01:48<08:39,  3.63s/it]

Iteration CLINTON_103: 0.9187 seconds


 24%|██▎       | 44/186 [01:49<06:35,  2.79s/it]

Iteration CLINTON_123_1: 0.8189 seconds


 24%|██▍       | 45/186 [01:50<05:05,  2.17s/it]

Iteration CLINTON_123_2: 0.7120 seconds
Exception repeat 1


 25%|██▍       | 46/186 [01:52<05:06,  2.19s/it]

Iteration CLINTON_130_1: 2.2540 seconds


 25%|██▌       | 47/186 [01:53<04:33,  1.97s/it]

Iteration CLINTON_130_2: 1.4382 seconds


 26%|██▌       | 48/186 [01:54<03:51,  1.67s/it]

Iteration CLINTON_147: 0.9940 seconds
Exception repeat 1


 26%|██▋       | 49/186 [01:56<03:49,  1.68s/it]

Iteration CLINTON_17: 1.6796 seconds
Exception repeat 1


 27%|██▋       | 50/186 [01:57<03:31,  1.56s/it]

Iteration CLINTON_176_1: 1.2792 seconds


 27%|██▋       | 51/186 [01:58<03:01,  1.35s/it]

Iteration CLINTON_176_2: 0.8544 seconds
Exception repeat 1


 28%|██▊       | 52/186 [02:00<03:36,  1.62s/it]

Iteration CLINTON_186: 2.2491 seconds


 28%|██▊       | 53/186 [02:01<03:03,  1.38s/it]

Iteration CLINTON_193: 0.8164 seconds


 29%|██▉       | 54/186 [02:02<02:35,  1.18s/it]

Iteration CLINTON_199_1: 0.7162 seconds


 30%|██▉       | 55/186 [02:03<02:20,  1.07s/it]

Iteration TRUMP_240_2: 0.8213 seconds


 30%|███       | 56/186 [02:04<02:21,  1.09s/it]

Iteration TRUMP_249: 1.1257 seconds
Exception repeat 1


### Evaulate 

- Run shell 

```shell

python3 eval_scripts/evaluation.py \
    --metric similarity \
    --input_path data_splits/validation.json \
    --submission_path output/elbaff_experiment/output_gpt-4o-mini-2024-07-18social_n2_Te_Sr.json \
    --threshold 0.6 

```

- OUTPUT : output/output_o3-mini_temperatureNA._eval_similarity_06.json



**Overall count**

| **Questions Labels** |  **#**  | **ratio** |
|:--------------------|-------:|---------:|
| useful               | **329** |  **0,59** |
| unhelpful            |      63 |      0,11 |
| Invalid              |       8 |      0,01 |
| Not able to evaluate |     158 |      0,28 |
| **Total**            | **558** |         1 |


**Overall count within each argument**

| **n/3 useful questions per arg** | **# of arguments** | **ratio** |
|:-------------------------------:|-------------------:|:---------:|
|                               0/3 |                 17 |      0,10 |
|                               1/3 |                 51 |      0,27 |
|                               2/3 |             **76** |      0,40 |
|                               3/3 |                 42 |      0,23 |
| **Total**                       |            **186** |         1 |

## Social Agents


In [ ]:
%pip show langchain-core

In [ ]:
import pandas as pd
import os

from social_agents import helper
from social_agents.agent_builder import   SocialAgentBuilder

EXPERIMENT_SETTINGS_FILE_PATH = "output/elbaff_experiment/experiment_settings.csv"
override = False
if not os.path.exists(EXPERIMENT_SETTINGS_FILE_PATH) or override:
    print("generating exp settings file")
    exps_df = helper.generate_experiment_settings()
    exps_df.to_csv(EXPERIMENT_SETTINGS_FILE_PATH, index=False)

experiment_settings = pd.read_csv(EXPERIMENT_SETTINGS_FILE_PATH)
experiment_settings.head()

# exp_row = experiment_settings.iloc[20].to_dict()
# exp_row
experiment_settings = experiment_settings.sort_values(by=["rounds", "number_of_agents"])

In [ ]:
import ast
import time
from social_agents import data_model
import ast
from tqdm import tqdm



temperature = 0.7
llm_name_short = "mistral24b_"
llm_name = llm_dic[llm_name_short]
def run_all_exp_settings(experiment_settings):
    all_done = False
    for _, row in tqdm(experiment_settings.iterrows(), total=len(experiment_settings)):
        exp_name = row["experiment_name"].format(llm_name= llm_name_short)
        if os.path.exists(f"{SocialAgentBuilder.ROOT_FOLDER}output_{exp_name}.json"):
            print("Experiment ", exp_name, "already done!")
            continue
        if True:##"easy_going" in list(ast.literal_eval(row["traits"])): #and len(list(ast.literal_eval(row["traits"]))) <3 and len(list(ast.literal_eval(row["strategies"]))) < 3:
            print("="*20)
            print(row.to_dict())
            print("EXPERIMENT NAME: ", exp_name)
            print("="*20)

            social_agent = SocialAgentBuilder(
                model_thread_id=row["thread_id"],
                llm_name = llm_name,
                llm_num = row["number_of_agents"],
                experiment_name= exp_name,
                temperature=temperature,
                collaborative_strategy=list(ast.literal_eval(row["strategies"])),
                agent_trait_lst=list(ast.literal_eval(row["traits"])))
            #display(Image(social_agent.graph.get_graph(xray=1).draw_mermaid_png()))

            social_agent.run_experiment(data_type = "validation", save= True)
            print(f'finished {exp_name}')
    all_done = True
    return all_done
delay_in_sec = 1
all_done = False
while True and not all_done: 
    try:###
        all_done = run_all_exp_settings(experiment_settings=experiment_settings)
    except Exception as e:
        all_done = False
        print(f"Exception caught: {e}, retrying in {delay_in_sec} seconds...")
        time.sleep(delay_in_sec)
